Update asciidoc plugins, config

starksm64 opened the pull request:
Addresses #558

manovotn commented on spec/pom.xml:
```diff
@@ -190,10 +190,10 @@
 					<attributes>
 						<license>${license-file}</license>
 						<revremark>${revremark}</revremark>
+						<source-highlighter>coderay</source-highlighter>
```
IIRC, these code highlighters only work if you have them locally installed when building the doc.
There is one exception to it - `highlight.js`.

So I have two questions:
* Would it be worth having `highlight.js` for HTML doc build and something else for PDF? So that local builds for anyone would have highlighting in HTML at least.
* Are we sure our Jenkins machines have this ruby gem installed? (for releases)

starksm64 replied:
I have tested this PR using the https://ci.eclipse.org/cdi/job/CDISpec job. There are no errors building the specs html/pdf and the code fragments look good in both. See https://ci.eclipse.org/cdi/job/CDISpec/8/ build artifacts. You have to download the html version for it to display correctly.

Ladicek commented:
I looked into updating Asciidoctor when prototyping specref and I did it like this.

1. Latest Asciidoctor Maven plugin doesn't include latest Asciidoctor. I added a new property for it:

```xml
...
<properties>
    <asciidoctor-maven.version>2.2.1</asciidoctor-maven.version>
    <asciidoctorj.version>2.5.2</asciidoctorj.version> <!-- <<<<< added line <<<<< -->
    <asciidoctorj-pdf.version>1.6.0</asciidoctorj-pdf.version>
   ...
</properties>
...
```

2. and a dependency to the Asciidoctor Maven plugin:

```xml
...
<plugin>
    <groupId>org.asciidoctor</groupId>
    <artifactId>asciidoctor-maven-plugin</artifactId>
    <version>${asciidoctor-maven.version}</version>
    <dependencies>
        <!-- >>>>> added dependency >>>>> -->
       <dependency>
           <groupId>org.asciidoctor</groupId>
            <artifactId>asciidoctorj</artifactId>
            <version>${asciidoctorj.version}</version>
        </dependency>
        <!-- <<<<< added dependency <<<<< -->
        <dependency>
            <groupId>org.asciidoctor</groupId>
            <artifactId>asciidoctorj-pdf</artifactId>
            <version>${asciidoctorj-pdf.version}</version>
        </dependency>
    </dependencies>
    ...
</plugin>
```

3. Asciidoctor Maven plugin has some breaking changes. These configuration changes were enough for me, though I didn't try rendering a PDF:

```xml
<configuration>
    <sourceDirectory>src/main/asciidoc</sourceDirectory> <!-- otherwise warnings are printed -->
    <sourceDocumentName>cdi-spec.asciidoc</sourceDocumentName>
    <attributes>
        <license>${license-file}</license>
        <revremark>${revremark}</revremark>
        <imagesdir>images</imagesdir>
        <source-highlighter>coderay</source-highlighter>
    </attributes>
</configuration>
```

starksm64 commented:
I have updated the plugin configuration to use the explicit asciidoctorj.version setting and added the addition configuration/attribute changes. The test build with this PR shows no errors/warnings:
https://ci.eclipse.org/cdi/job/CDISpec/8/console

Ladicek commented:
I vaguely recall seeing a `3.0` somewhere else too, but that should be fixed in another PR.
